### ISSUE TITLE
AssertionError in xlsx.py

### ISSUE THREAD

**Issue by EleventyOne** [python-excel/xlrd#56]
xlrd: 0.9.2
python: 3.3 (windows-64)

I'm trying to read in an .xlsx file (containing 17,000+ rows) and receiving an AssertionError. Here is the call stack:

====

Traceback (most recent call last):
  File "C:\Users\Me\Documents\python\test\reading_excel_files.py", line 8, in <module>
    book = xlrd.open_workbook(FILENAME)
  File "C:\Python33\lib\site-packages\xlrd\__init__.py", line 416, in open_workbook
    ragged_rows=ragged_rows,
  File "C:\Python33\lib\site-packages\xlrd\xlsx.py", line 757, in open_workbook_2007_xml
    x12sheet.process_stream(zflo, heading)
  File "C:\Python33\lib\site-packages\xlrd\xlsx.py", line 520, in own_process_stream
    self_do_row(elem)
  File "C:\Python33\lib\site-packages\xlrd\xlsx.py", line 553, in do_row
    assert 0 <= self.rowx < X12_MAX_ROWS
AssertionError

====

Unfortunately, the .xlsx file that is causing the AssertionError contains confidential information on research subjects and so I am not able to post any of it. Through my own investigation:

(1) self.rowx is -1, and so that's causing the AssertionError. 
(2) If I add code to set self.rowx to 0 if it's < 0 (at the end of the else block before the assert), the file SEEMS to be read fine
(3) If I open the .xlsx file in OpenOffice and save it as a new .xls file, the new file SEEMS to be read in fine without the added code

This is the first time I've ever used xlrd, so I'm hesitant to say (in points 2 and 3 above) that things ARE working fine. I've barely even started trying to process my file and so it's entirely possible that I'll run into other problems later on. Nevertheless, since this problem was preventing me from moving ahead at all with my work, I needed to add the hack in point 2 in order to progress (converting all the files to .xls is not a feasible, long-term solution)

Hopefully the information I've provided here can be somewhat helpful in identifying the problem. Although I do realize a source .xlsx file would be most helpful, that's simply not an option here.

Edit: Using either the added code hack (point 2) or simply converting the file (point 3), I can confirm that the following vars/methods called on my workbook seem to report the correct information: .name, .nrows, .ncols, .cell_value(), .row()

Edit2: Other than that assert I encountered, everything else seems to work fine for me in this workbook. This is a great module, thanks for all the hard work.

Edit3: I've tried loading two other .xlsx files containing different data that were exported from the same database, and I've received the exact same assert in the same location for the same reason. These other files had significantly less rows (5000-). So I went looking for sample .xlsx files on the net (OpenOffice can't save as that format), downloaded a bunch, and was able to open them all successfully. So obviously there's something special about these particular .xlsx files that's causing the trouble. Frustrating.

**Comment by cjw296** [comment]
Should be fixed in 1.0.1

**Comment by vadella** [comment]
I have the same problem with an xlsx-file exported from SAP (netweaver) as `Office 2007 XLSX Format` which I can't share. I tried exporting it as one of the other formats, but those were XML and MHTML-formats, so I haven't found a way to read them yet.
I could read the file in MS Excel without problem, but the same assertion error popped up when I tried to read it with pandas/xlrd.

My `sheet1.xml` was not empty and contained this somewhere.
```
<sheetData>
  <row r="0" spans="">
    <c r="A1" s="1" t="s">
    ...
```

When I changed the `0` to `1` and rezipped everything, the outcome looked exactly the same in Excel, but posed no problems for pandas/xlrd. 

Since my first row actually contains data, I don't know whether the fix suggested here will work for my case.

version info:
Python 3.5.2 (Anaconda)
pandas: 0.19.1
xlrd: 1.0.0

**Comment by cjw296** [comment]
@vadella  - So how about reading my comment and using 1.0.1 instead of 1.0.0?

**Comment by vadella** [comment]
@cjw296 - I use 1.0.0 because that's the version available both on pypi and conda, so it's difficult for me to test it out. And my knowledge of the xlrd codebase isn't sufficient to estimate what effect the changes in [patch #107](https://github.com/python-excel/xlrd/pull/107/files) have if there is indeed something in the row with `<row r="0"...`

**Comment by ulidtko** [comment]
Version 1.1.0 successfully fixed the AssertionError for me. Thanks!

**Comment by teodoragoran** [comment]
> Version 1.1.0 successfully fixed the AssertionError for me. Thanks!

Are you sure? I have tried with version 1.1.0 and I am still able to reproduce this issue.

**Comment by mgab** [comment]
With version 1.1.0 I encountered the issue. In fact, I can also reproduce it using the [example file](https://github.com/python-excel/xlrd/issues/56#issuecomment-21462516) provided by @EleventyOne.

Similarly, adding [the print line](https://github.com/python-excel/xlrd/issues/56#issuecomment-19161744) `print(self.rowx, repr(row_number), explicit_row_number)` suggested by @sjmachin gets the same results: `-1 '0' 1`. I tried it both with python 2.7 and 3.6

@cjw296, I could not find any version tagged 1.0.1, nor in github neither in pypi. Thus I could not try it with that version. Maybe there has been some regression to the problematic behaviour (due to lack of hands, no tests were added to the PR)? In any case, if I'm not the only one that can reproduce the bug, it probably should be reopened...
